helios-services/src/main/java/com/spotify/helios/master/MasterService.java

```diff
@@ -308,13 +316,40 @@ private void logBanner() {
    * @return A zookeeper client.
    */
   private ZooKeeperClient setupZookeeperClient(final MasterConfig config) {
+    ACLProvider aclProvider = null;
+    List<AuthInfo> authorization = null;
+
+    if (config.isZooKeeperEnableAcls()) {
```

Comment:
If you have a good idea of how to do it, please share. I can move it to another place to reduce clutter, but I can't really figure out a great way to reduce duplication.